src/main/java/org/ethereum/beacon/discovery/util/RlpUtil.java

```diff
@@ -118,23 +138,33 @@ public static DecodedList decodeFirstList(Bytes data) {
    *
    * @throws RlpDecodeException if this rlp doesn't represent a single list
    */
-  public static RlpList decodeSingleList(Bytes rlp) throws RlpDecodeException {
-    RlpType item = decodeSingleItem(rlp);
+  public static List<RlpType> decodeSingleList(Bytes rlp) throws RlpDecodeException {
+    return asList(decodeSingleItem(rlp));
+  }
+
+  public static List<RlpType> asList(RlpType item) throws RlpDecodeException {
     if (!(item instanceof RlpList)) {
-      throw new RlpDecodeException("Expected RLP list, but got a string from bytes: " + rlp);
+      throw new RlpDecodeException("Expected RLP list, but got a string: " + item);
     }
-    return (RlpList) item;
+    return ((RlpList) item).getValues();
   }
 
-  /**
-   * Decodes strictly one bytes string item from rlp bytes
-   *
-   * @throws RlpDecodeException if this rlp doesn't represent a single bytes string
-   */
-  public static Bytes decodeSingleString(Bytes rlp, int expectedSize) throws RlpDecodeException {
-    Bytes ret = decodeSingleString(rlp);
-    if (ret.size() != expectedSize) {
-      throw new RlpDecodeException("Expected string size doesn't match: " + rlp);
+  public static int asInteger(RlpType item) throws RlpDecodeException {
+    Bytes bytes = asString(item, maxSize(4));
+    long l = bytes.toLong();
+    if (l > Integer.MAX_VALUE) {
```

Comment:
Shouldn't we check against `Integer.MIN_VALUE` as well?

Reply:
`4 bytes` to `Long` would be definitely positive